### PR TITLE
Docker image build 

### PR DESCRIPTION
**Pull request by Legion112**
Add docker image build
https://github.com/vimeo/psalm/discussions/6602#discussioncomment-2196704

- [ ] add GitAction automatic build for image
- [ ] Refactor to use phar to minimize image size
- [ ] setup credentials for psalm on docker hub

### CODE REVIEW

File: entrypoint [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
```
Comment (weirdan) [inline entrypoint:1]: Please move docker stuff into its own folder, e.g. `docker`. It'll be easier to export-ignore it in package distribution.

**Comment by Legion112** [comment]
Move to another request. Have some problem with branch name.
https://github.com/vimeo/psalm/pull/7701

File: entrypoint (hunk 1 above)
Comment (Legion112) [reply entrypoint:1]: This one impossible to move. Must be in the root. 
.dockerignore

Comment (Legion112) [reply entrypoint:1]: https://github.com/vimeo/psalm/pull/7701/commits/957001bfdec4bef2855fdc19dbd16fbe8920abf8

**Pull request closed without merging**